src/components/admin/event/EventDetailsForm/index.tsx

```diff
@@ -0,0 +1,320 @@
+import { SubmitHandler, useForm } from 'react-hook-form';
+import style from './style.module.scss';
+
+interface IProps {
+  editing?: boolean;
+  defaultData?: Partial<PublicEvent>;
+}
+
+const EventDetailsForm = (props: IProps) => {
+  const { editing, defaultData } = props;
+  const router = useRouter();
+  const initialValues: Partial<PublicEvent> = {
+    title: defaultData?.title || '',
+    committee: defaultData?.committee || 'General',
+    location: defaultData?.location || '',
+    pointValue: defaultData?.pointValue || 10,
+    start: defaultData?.start || '',
+    end: defaultData?.end || '',
+    attendanceCode: defaultData?.attendanceCode || '',
+    description: defaultData?.description || '',
+    cover: defaultData?.cover,
+    uuid: defaultData?.uuid,
+  };
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    reset,
+    formState: { errors },
+  } = useForm<Event>({ defaultValues: initialValues });
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const [upcomingEvents, setUpcomingEvents] = useState<NotionEventPreview[]>([]);
+
+  // Load in the Notion event options for NotionAutofill using Klefki.
+  useEffect(() => {
+    setLoading(true);
+    KlefkiAPI.getFutureEventsPreview().then(notionUpcomingEvents => {
+      setUpcomingEvents(notionUpcomingEvents);
+      setLoading(false);
+    });
+  }, []);
+
+  const setFieldsFromAutofill = (data: NotionEventDetails) => {
+    const { checkin, title, description, start, end, location, community } = data;
+    setValue('attendanceCode', checkin, { shouldValidate: true });
+    setValue('title', title, { shouldValidate: true });
+    setValue('description', description, { shouldValidate: true });
+    setValue('start', DateTime.fromISO(start).toFormat("yyyy-MM-dd'T'HH:mm") ?? '', {
+      shouldValidate: true,
+    });
+    setValue('end', DateTime.fromISO(end).toFormat("yyyy-MM-dd'T'HH:mm") ?? '', {
+      shouldValidate: true,
+    });
+    setValue('location', location, { shouldValidate: true });
+    setValue('committee', community);
+  };
+
+  const resetForm = () => reset(initialValues);
+
+  const createEvent: SubmitHandler<FillInLater> = formData => {
+    setLoading(true);
+    const { start: isoStart, end: isoEnd, ...event } = formData;
+
+    const start = new Date(isoStart).toISOString();
+    const end = new Date(isoEnd).toISOString();
+
+    const AUTH_TOKEN = CookieService.getClientCookie(CookieType.ACCESS_TOKEN);
+
+    AdminEventManager.createNewEvent({
+      token: AUTH_TOKEN,
+      event: {
+        ...event,
+        start,
+        end,
+      },
+      cover: event.cover[0],
+      onSuccessCallback: event => {
+        setLoading(false);
+        showToast('Event Created Successfully!', '', [
+          {
+            text: 'View Live Event Page',
+            onClick: () => router.push(`https://acmucsd.com/events/${event.uuid}`),
+          },
+        ]);
+        router.push(`/admin/event`);
```

Comment:
nitpick: should this be in normal quotes, or in config.ts?

Reply:
ah good catch, i'll update all the route stuff to use config.ts